applications/crossbar/src/modules/cb_rates.erl

```diff
@@ -435,3 +446,27 @@ save_processed_rates(Context, Count) ->
                   _ = crossbar_doc:save(Context, [{'publish_doc', 'false'}]),
                   lager:debug("saved up to ~b docs (took ~b ms)", [Count, wh_util:elapsed_ms(Now)])
           end).
+
+-spec rate_for_number(ne_binary(),ne_binary()) -> cb_context:context().
+rate_for_number(Context, Phonenumber) ->
+    Resp = case wh_amqp_worker:call([{<<"To-DID">>, wnm_util:normalize_number(Phonenumber)}| wh_api:default_headers(?APP_NAME, ?APP_VERSION)]
+                                    ,fun wapi_rate:publish_req/1
+                                    ,fun wapi_rate:resp_v/1
+                                    ,10000) of
+               {'ok', Rate} ->  
+                   Routines = [fun(J) -> wh_json:set_value(<<"Base-Cost">>, wh_util:to_binary(wh_util:to_float(wh_json:get_value(<<"Base-Cost">>, J))/10000), J) end
+                               ,fun(J) -> wh_json:set_value(<<"Rate">>, wh_util:to_binary(wh_util:to_float(wh_json:get_value(<<"Rate">>, J))/10000), J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Event-Category">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Event-Name">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"App-Name">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"App-Version">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Msg-ID">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Node">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Call-ID">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Pvt-Cost">>, J) end
+                               ,fun(J) -> wh_json:delete_key(<<"Update-Callee-ID">>, J) end
```

Comment:
maybe you can create a macro at the top of module with the keys you want to delete and just call delete_keys with that macro, or create a macro with keys you want to include and call wh_json:map